lib/filewatch/discoverer.rb

```diff
@@ -48,20 +48,17 @@ def can_exclude?(watched_file, new_discovery)
     end
 
     def discover_files(path)
-      globbed = Dir.glob(path)
-      globbed = [path] if globbed.empty?
-      logger.debug("Discoverer found files, count: #{globbed.size}")
-      globbed.each do |file|
+      fileset = Dir.glob(path).select{|f| File.file?(f) && !File.symlink?(f)}
+      logger.debug("Discoverer found files, count: #{fileset.size}")
+      fileset.each do |file|
         logger.debug("Discoverer found file, path: #{file}")
         pathname = Pathname.new(file)
-        next unless pathname.file?
-        next if pathname.symlink?
         new_discovery = false
         watched_file = @watched_files_collection.watched_file_by_path(file)
         if watched_file.nil?
-          logger.debug("Discoverer discover_files: #{path}: new: #{file} (exclude is #{@exclude.inspect})")
           new_discovery = true
           watched_file = WatchedFile.new(pathname, pathname.stat, @settings)
+          logger.debug("Discoverer new discovery: #{watched_file.path}, inode: #{watched_file.sincedb_key.inode}, discover_files: #{path} (exclude is #{@exclude.inspect})")
```

Comment:
minor: you are using the structured logs notations in most places, maybe here too?

Reply:
I started to change the logging to structured but its a PR all of its own.